File: netCore/dojodachi/wwwroot/js/dachi.js
$(document).ready(function () {
    $(".feed").click(function(){
        $.ajax({
            url: "feed",
            method: "GET",
            success: function(response){
                console.log(response.meal);
                console.log(response.fullness);
                if(response.meal !== undefined){
                    $('.meal').text(response.meal);
                    $('.fullness').text(response.fullness);
                    if(response.win === true){
                        $('.feed').css("visibility", "hidden");
                        $('.play').css("visibility", "hidden");
                        $('.work').css("visibility", "hidden");
                        $('.sleep').css("visibility", "hidden");
                        $('.restart').css("visibility", "visible");
                        $(".card-text").text("You Won!! Restart?");
                    } else {    
                        if(response.gain === 0){
                            $(".card-text").text("Unhappy! Gained No Fullness");
                        } else if (response.gain === -1) {
                            $(".card-text").text("You don't even have a meal to feed");
                        } else {
                            $(".card-text").text("Good! Gained " + response.gain + " Fullness, -1 Meal");
                        }
                    }
                }
            }
        })
    })
    $(".play").click(function(){
        $.ajax({
            url: "play",
            method: "GET",
            success: function(response){
                $('.energy').text(response.energy);
                $('.happiness').text(response.happiness);
                if(response.win === true){
                    $('.feed').css("visibility", "hidden");
                    $('.play').css("visibility", "hidden");
                    $('.work').css("visibility", "hidden");
                    $('.sleep').css("visibility", "hidden");
                    $('.restart').css("visibility", "visible");
                    $(".card-text").text("You Won!! Restart?");
                } else {
                    if(response.gain === 0){
                        $(".card-text").text("Unhappy! Gained No Happiness");
                    } else if (response.gain === -1) {
                        $(".card-text").text("You don't even have energy to play");
                    } else {
                        $(".card-text").text("Good! Gained " + response.gain + " Happiness, -5 Energy");
                    }
                }
            }
        })
    })
    $(".work").click(function(){
        $.ajax({
            url: "work",
            method: "GET",
            success: function(response){
                $('.energy').text(response.energy);
                $('.meal').text(response.meal);
                if(response.gain === -1){
                    $(".card-text").text("You don't have energy to work. Sleep?");
                } else {
                    $(".card-text").text("Good! Gained " + response.gain + " Meal, -5 Energy");
                }

            }
        })
    })
    $(".sleep").click(function(){
        $.ajax({
            url: "sleep",
            method: "GET",
            success: function(response){
                $('.energy').text(response.energy);
                $('.happiness').text(response.happiness);
                $('.fullness').text(response.fullness);
                if(response.lose === true){
                    $('.feed').css("visibility", "hidden");
                    $('.play').css("visibility", "hidden");
                    $('.work').css("visibility", "hidden");
                    $('.sleep').css("visibility", "hidden");
                    $('.restart').css("visibility", "visible");
                    $(".card-text").text("You Lost.....!! Restart?");
                } else if (response.won === true){
                    $('.feed').css("visibility", "hidden");
                    $('.play').css("visibility", "hidden");
                    $('.work').css("visibility", "hidden");
                    $('.sleep').css("visibility", "hidden");
                    $('.restart').css("visibility", "visible");
                    $(".card-text").text("You Won!! Restart?");
                } else {    
                    $(".card-text").text("Good! Gained " + response.gain + " Energy, -5 Happiness & Fullness");
                }
                

            }
        })
    })
});

File: netCore/dojodachi/Dachi.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace dojodachi {

    public class Dachi {

        public int happiness { get; set; }
        public int fullness { get; set; }
        public int energy { get; set; }
        public int meal { get; set; }
        public Random random = new Random();
        public int failure = 1;

        public Dachi() {
            happiness = 20;
            fullness = 20;
            energy = 50;
            meal = 3;
        }

        public int Feed() {
            if(this.meal > 0){
                if(failure == random.Next(0,4)){
                    this.meal -= 1;
                } else {
                    int gain = random.Next(5, 11);
                    this.fullness += gain;
                    this.meal -= 1;
                    return gain;
                }
            } else {
                return -1;
            }
            return 0;
        }

        public int Play() {
            if(this.energy >= 5){
                if(failure == random.Next(0,4)){
                    this.energy -= 5;
                } else {
                    int gain = random.Next(5, 11);
                    this.happiness += gain;
                    this.energy -= 5;
                    return gain;
                }
            } else {
                return -1;
            }
            return 0;
        }

        public int Work() {
            if(this.energy >= 5){
                int gain = random.Next(1, 4);
                this.energy -= 5;
                this.meal += gain;
                return gain;
            } else {
                return -1;
            }
        }

        public int Sleep() {
            this.energy += 15;
            this.fullness -= 5;
            this.happiness -= 5;
            return 15;
        }

        public bool CheckWin(){
            if(this.fullness >= 100 && this.energy >= 100 && this.happiness >= 100) {
                return true;
            } 
            return false;
        }

        public bool CheckLose() {
            if(this.fullness <= 0 || this.happiness <= 0) {
                return true;
            }
            return false;
        }

    }
}
